### DISCUSSION TITLE
Evaluate the introduction of a radashi.js dependency

### DISCUSSION THREAD

**Discussion by dgdavid** [agama-project/agama#1601] [Ideas]
First of all, let me clarify that I'm aware of the disadvantages of excessive reliance on dependencies and did not forget the https://en.wikipedia.org/wiki/Npm_left-pad_incident

Moreover, when it comes to dependencies, I always think of the [picture](https://www.xkcd.com/2347/) below.

![dependency](https://github.com/user-attachments/assets/65e4c4e7-07b4-4a44-93b0-e1a518819d6a)

However, I realized recently that we could rely on https://radashi.js.org/ to stop writing the utilities ourselves. It already provides our `noop`, `isEmpty` (recently added by myself), `isObject`, `isObjectEmpty` through its `isEmpty`, `partition` (which they call `fork`), etc.

We can benefit from its use by writing and maintaining less code, documentation, and tests. Rather than reinventing the wheel, we could focus on Agama features instead :P

Please, note that it is built for TypeScript and fully tree-shakeable.

**Comment by imobachgs** [comment]
In my opinion, it makes sense. And if something goes wrong, we could replace the helpers with our own implementations again. But I understand that this library is properly tested and it will save us of writing trivial code (with its share of bugs).

**Reply by dgdavid** [reply]
> And if something goes wrong, we could replace the helpers with our own implementations again.

We could even _keeps_ a copy of the library in a automatically syncronized repo, just in case.

> But I understand that this library is properly tested

Yes it is.

>  and it will save us of writing trivial code (with its share of bugs).

And even allow us to contribute with something missing, if needed.

**Comment by ancorgs** [comment]
I do not oppose.

**Comment by dgdavid** [comment]
Status update: I've been working a bit on this. I'll put in on hold for a couple of weeks, at least until v 12.2 is published.

**Comment by dgdavid** [comment]
For the record, today I've created a PR for going ahead with this if finally accepted. https://github.com/agama-project/agama/pull/2412

**Reply by dgdavid** [reply]
And now merged to `master` branch.
